adapter/serviceControl/serviceController_test.go

```diff
@@ -0,0 +1,180 @@
+	}
+}
+
+func TestAspect_Close(t *testing.T) {
+	a := new(aspect)
+	if err := a.Close(); err != nil {
+		t.Errorf("Close() => unexpected error: %v", err)
+	}
+}
+
+func TestAspect_Log(t *testing.T) {
+	a := &aspect{
+		testServiceName,
+		new(servicecontrol.Service),
+		test.NewEnv(t).Logger(),
+	}
+
+	l := adapter.LogEntry{LogName: "istio_log", TextPayload: "text payload", Timestamp: "2017-Jan-09", Severity: adapter.Info}
+
+	tests := []struct {
+		input []adapter.LogEntry
+		want  servicecontrol.ReportRequest
+	}{
+		{[]adapter.LogEntry{l},
+			servicecontrol.ReportRequest{
+				Operations: []*servicecontrol.Operation{
+					{
+						OperationId:   "test_operation",
+						OperationName: "reportLogs",
+						StartTime:     "2017-07-01T10:10:05.000000002Z",
+						EndTime:       "2017-07-01T10:10:05.000000002Z",
+						LogEntries: []*servicecontrol.LogEntry{
+							{
+								Name:        "istio_log",
+								Severity:    "INFO",
+								TextPayload: "text payload",
+								Timestamp:   "2017-Jan-09",
+							},
+						},
+						Labels: map[string]string{"cloud.googleapis.com/location": "global"},
+					},
+				},
+			},
+		},
+	}
+
+	for _, v := range tests {
+		r := a.log(v.input, time.Date(2017, time.July, 1, 10, 10, 5, 2, time.UTC), "test_operation")
+		if !reflect.DeepEqual(*r, v.want) {
+			t.Errorf("log(%+v) => %v, want %v", v.input, spew.Sdump(*r), spew.Sdump(v.want))
+		}
+	}
+}
+
+func TestAspect_Record(t *testing.T) {
+	a := &aspect{
+		testServiceName,
+		new(servicecontrol.Service),
+		test.NewEnv(t).Logger(),
+	}
+
+	c := int64(123)
+	v := adapter.Value{
+		Definition: &adapter.MetricDefinition{
+			Name: "request_count",
+			Kind: adapter.Counter,
+		},
+		Labels:      make(map[string]interface{}),
+		StartTime:   time.Date(2017, time.June, 30, 18, 10, 5, 2, time.UTC),
+		EndTime:     time.Date(2017, time.June, 30, 18, 10, 30, 2, time.UTC),
+		MetricValue: c,
+	}
+
+	v.Labels["response_code"] = 500
+
+	tests := []struct {
+		input []adapter.Value
+		want  servicecontrol.ReportRequest
+	}{
+		{[]adapter.Value{v},
+			servicecontrol.ReportRequest{
+				Operations: []*servicecontrol.Operation{
+					{
+						OperationId:   "test_operation",
+						OperationName: "reportMetrics",
+						StartTime:     "2017-07-01T10:10:05.000000002Z",
+						EndTime:       "2017-07-01T10:10:05.000000002Z",
+						MetricValueSets: []*servicecontrol.MetricValueSet{
+							{
+								MetricName: "serviceruntime.googleapis.com/api/producer/request_count",
+								MetricValues: []*servicecontrol.MetricValue{
+									{
+										Labels: map[string]string{
+											"/response_code": "500",
+										},
+										StartTime:  "2017-06-30T18:10:05.000000002Z",
+										EndTime:    "2017-06-30T18:10:30.000000002Z",
+										Int64Value: &c,
+									},
+								},
+							},
+						},
+						Labels: map[string]string{"cloud.googleapis.com/location": "global"},
+					},
+				},
+			},
+		},
+	}
+
+	for _, v := range tests {
```

Comment:
note: you've reused `v` here. you may prefer to rename one of the two places you've used it.